model/modeljson/internal/document.go

```diff
@@ -68,25 +68,21 @@ type Document struct {
 	Process     *Process     `json:"process,omitempty"`
 	Event       *Event       `json:"event,omitempty"`
 
-	Timestamp  Time        `json:"@timestamp"`
 	DataStream *DataStream `json:"data_stream,omitempty"`
 	Message    string      `json:"message,omitempty"`
 	DocCount   uint64      `json:"_doc_count,omitempty"`
+	Timestamp  Time        `json:"@timestamp"`
 }
 
-type Time time.Time
+type Time uint64
 
 func (t Time) MarshalFastJSON(w *fastjson.Writer) error {
 	w.RawByte('"')
-	w.Time(time.Time(t), timestampFormat)
+	w.Time(time.Unix(0, int64(t)).UTC(), timestampFormat)
```

Comment:
Should we use PBTimestampToTime here?

Alternatively I think we could leave `modeljson` alone, and convert to time.Time in the caller.